stream_alert/rule_processor/rules_engine.py

```diff
@@ -346,52 +349,113 @@ def process(cls, input_payload):
         alerts = []
         payload = copy(input_payload)
 
-        rules = [rule_attrs for rule_attrs in cls.__rules.values()
+        rules = [rule_attrs for rule_attrs in self.__rules.values()
                  if rule_attrs.logs is None or payload.log_source in rule_attrs.logs]
 
         if not rules:
             LOGGER.debug('No rules to process for %s', payload)
             return alerts
 
+        # store normalized records for future process in Threat Intel
+        normalized_records = []
+
         for record in payload.records:
             for rule in rules:
                 # subkey check
-                has_sub_keys = cls.process_subkeys(record, payload.type, rule)
+                has_sub_keys = StreamRules.process_subkeys(record, payload.type, rule)
```

Comment:
why call `StreamRules` vs `self`? there's some inconsistency in the subsequent blocks below

Reply:
Good catch! Thanks!